File: python/Day25_RunningTimeAndComplexity.py
import sys
# 참고 https://myjamong.tistory.com/139

def check_prime(num):
    if num == 1:
        return False
    elif num <= 3:
        return True
    else:
        end_num = int(num ** 0.5) + 1
        for i in range(2, end_num):
            if num % i == 0:
                return False
        return True

n = int(sys.stdin.readline())

for i in range(n):
    num = int(sys.stdin.readline())

    if check_prime(num):
        print("Prime")
    else:
        print("Not prime")

File: scala/Day25_RunningTimeAndComplexity.scala
object Solution {

    def check_prime(num: Int) : Boolean = {
        if(num == 1)
            return false
        else if(num <= 3)
            return true
        else {
            var end_num = math.sqrt(num).toInt + 1
            for (i <- 2 until end_num) {
                if(num % i == 0)
                    return false
            }
            return true
        }
    }

    def main(args: Array[String]) {
        var n = scala.io.StdIn.readInt()

        for(i <- 0 until n) {
            var num = scala.io.StdIn.readInt()
            if (check_prime(num))
                println("Prime")
            else
                println("Not prime")
        }
    }
}